src/Microsoft.DocAsCode.Build.OverwriteDocuments/OverwriteDocumentModelCreater.cs

```diff
@@ -3,13 +3,198 @@
+                    OPathSegments = OverwriteUtility.ParseOPath(content.PropertyName);
+                }
+                catch (ArgumentException ex)
+                {
+                    throw new MarkdownFragmentsException(ex.Message, content.PropertyNameSource.Line);
+                }
+
+                AppendNewObject(OPathSegments, content.PropertyNameSource, content.PropertyValue, contentsMetadata);
+            }
+
+            return contentsMetadata;
+        }
+
+        private static void AppendNewObject(List<OPathSegment> OPathSegments, Block codeHeaderBlock, List<Block> propertyValue, Dictionary<string, object> contentsMetadata)
+        {
+            var objectValue = contentsMetadata;
+            var leftSegments = new List<OPathSegment>(OPathSegments);
+            foreach (var segment in OPathSegments)
+            {
+                if (objectValue.ContainsKey(segment.SegmentName))
+                {
+                    if (!string.IsNullOrEmpty(segment.Key))
+                    {
+                        var listObject = objectValue[segment.SegmentName] as List<Dictionary<string, object>>;
+                        if (listObject != null && listObject.Count > 0)
+                        {
+                            var goodItems = (from item in listObject
+                                where item.ContainsKey(segment.Key) &&
+                                      item[segment.Key].ToString().Equals(segment.Value.ToString(),
+                                          StringComparison.OrdinalIgnoreCase)
+                                select item).ToList();
+                            if (goodItems.Count > 0)
+                            {
+                                objectValue = goodItems.First();
+                                leftSegments.Remove(segment);
+                            }
+                            else
+                            {
+                                ((List<Dictionary<string, object>>) objectValue[segment.SegmentName]).Add(
+                                    ((List<Dictionary<string, object>>) ((Dictionary<string, object>) CreateObject(leftSegments, propertyValue))[segment.SegmentName])
+                                    .First());
+                                return;
+                            }
+                        }
+                        else
+                        {
+                            // Throw exception if there are two OPaths like:
+                            // A/B
+                            // A[c=d]/C
+                            var sameSegments = new List<OPathSegment>(OPathSegments);
+                            sameSegments.RemoveRange(OPathSegments.IndexOf(segment), OPathSegments.Count - OPathSegments.IndexOf(segment));
+                            throw new MarkdownFragmentsException(
+                                $"OPath {OPathSegments.Select(o => o.OriginalSegmentString).Aggregate((a, b) => a + "/" + b)} " +
```

Comment:
Add before some user friendly error message, e.g. `A is expected to be an object with "A/B", however it is used as an array in L 10 with "A[c=d]/c" syntax`